orderbook/src/api/get_fee_and_quote.rs

```diff
@@ -0,0 +1,238 @@
+use crate::fee::{MinFeeCalculating, MinFeeCalculationError};
+use anyhow::Result;
+use chrono::{DateTime, Utc};
+use ethcontract::{H160, U256};
+use model::h160_hexadecimal;
+use model::{order::OrderKind, u256_decimal};
+use serde::{Deserialize, Serialize};
+use shared::{
+    conversions::{big_int_to_u256, U256Ext},
+    price_estimate::{PriceEstimating, PriceEstimationError},
+};
+use std::convert::Infallible;
+use std::sync::Arc;
+use warp::{hyper::StatusCode, reply, Filter, Rejection, Reply};
+
+#[derive(Deserialize)]
+struct Query {
+    #[serde(with = "h160_hexadecimal")]
+    sell_token: H160,
+    #[serde(with = "h160_hexadecimal")]
+    buy_token: H160,
+    // For sell orders in sell token, for buy orders in the buy token.
+    #[serde(with = "u256_decimal")]
+    in_amount_before_fees: U256,
+    kind: OrderKind,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Response {
+    expiration_date: DateTime<Utc>,
+    #[serde(with = "u256_decimal")]
+    min_fee: U256,
+    // For sell orders in buy token, for buy orders in the sell token.
+    #[serde(with = "u256_decimal")]
+    out_amount_before_fees: U256,
+    // For buy orders this is the same as out_amount_before_fees. For sell orders this is less
+    // than out_amount_before_fees because the fee is first deducted from the sell amount before
+    // applying the price.
+    #[serde(with = "u256_decimal")]
+    out_amount_after_fees: U256,
```

Comment:
Nice names. Hopefully this kill any ambiguity

Reply:
EDIT: Now im not sure 😅, will for buy orders be actually really after fees? My understanding is that is not.

Reply:
As the comment says, for buy orders the amount before and after fee is the same.
The reason is that for buy orders the fee is in the sell token but we don't know how much the user has in the sell token. We only know they want want to buy x amount in the buy token. So what is returned is the fee they need to set and the sell amount they need set, but there is nothing to deduct. Does that make sense?

Reply:
Yep, it does. It's hard to find infalible names when we play with fees, prices, and buy/sell, but I think is good enough